bpo-20104: Remove posix_spawn from 3.7

pablogsal opened the pull request:
<!-- issue-number: bpo-20104 -->
https://bugs.python.org/issue20104
<!-- /issue-number -->

pablogsal committed:
Remove posix_spawn

gpshead reviewed (commented):
this looks straight forward.

I suggest dropping the changes to aclocal.m4, configure.ac, configure, pyconfig.h.in that remove the check for the API.  It is fine for us to leave the autoconf script in 3.7 testing for the posix_spawn function.

gpshead committed:
Add a news entry.

and reference the removal from the old news entry where it was added.

gpshead committed:
Undo the autoconf related changes.

We can keep the configure test for posix_spawn, we'll need it in the
future and having it in pyconfig.h will be convenient in case any
3.7 extension module wants to use it.
